### DISCUSSION TITLE
Add an option to pull environment variables from a specific Environment and Git Branch

### DISCUSSION THREAD

**Discussion by robin-ln** [vercel/vercel#8528] [Ideas] [answered]
Currently we can Add a new variable for a specific Environment and Git Branch with this commande : 

```
$ vercel env add <name> <production | preview | development> <gitbranch>
$ vercel env add DB_PASS preview feat1
```

But after we can't pull the variables for a specific Environment and Git Branch.

My proposal is to add an option to do so, for example:

```
vercel pull --environment preview --gitbranch staging
```

**Comment by giancarlosisasi** [comment]
+1 🙏🏼

**Comment by amyegan** [comment, marked as answer]
Good news! The `--git-branch` option is available on `vc pull` and `vc env pull` in [vercel@28.9.0](https://github.com/vercel/vercel/releases/tag/vercel%4028.9.0). 

Now you can use `vercel pull --environment=preview --git-branch=staging` to get preview environment variables specifically for the staging branch. 🎉

**Reply by dakdevs** [reply]
I'm using `vercel@28.10.1` and it does not work. It doesn't show up in the `-h` either.
```
Vercel CLI 28.10.1

  ▲ vercel pull [project-path]

 Options:

    -h, --help                     Output usage information
    -A FILE, --local-config=FILE   Path to the local `vercel.json` file
    -Q DIR, --global-config=DIR    Path to the global `.vercel` directory
    -d, --debug                    Debug mode [off]
    --environment [environment]    Deployment environment [development]
    -y, --yes                      Skip questions when setting up new project using default scope and settings
```

![image](https://user-images.githubusercontent.com/12757641/209601766-c4b5b4a4-0407-4ab5-9d10-8b232c9f7123.png)
It exists here.

![image](https://user-images.githubusercontent.com/12757641/209601833-37a5a69d-8985-437d-8953-159593b6600a.png)
```
yarn vercel pull --yes --environment=preview --git-branch="feature/tre-36-update-github-actions" --token=$VERCEL_TOKEN
```
I run this command.

Then when using `vercel build` it does not include the branch specific env var.

It doesn't show up in the generated `.env.preview.local` file either.

**Reply by nathanielpomianowski** [reply]
Why not add this as an option to vercel deploy as well?
